src/components/ButtonLink/ButtonLink.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { ButtonProps, buttonStyles, iconSizeMap } from "../Button";
+import { Icon, IconName } from "../Icon";
+
+export interface ButtonLinkProps<T extends React.ElementType>
+  extends ButtonProps {
+  as?: T;
+  children?: React.ReactNode;
+  href: string;
+  iconLeft?: IconName;
+  iconRight?: IconName;
+  target?: string;
+}
```

Comment:
Let's not add new props to `ButtonLink`, and let's not extend `ButtonProps`. `ButtonLinkProps` should only be this: 

```suggestion
export type ButtonLinkProps<T extends React.ElementType> = React.PropsWithChildren<
   React.ComponentPropsWithoutRef<T> & {
      as?: T
   }
```

Reply:
Implementing it!